### PR TITLE
Fix ios13 registration issue

### PR DESCRIPTION
**Pull request by iTOYS**

### CODE REVIEW

File: src/WindowsAzureMessaging/WindowsAzureMessaging/SBNotificationHub.m [hunk 1]
```diff
@@ -79,11 +79,14 @@ - (SBNotificationHub*) initWithConnectionString:(NSString*) connectionString not
...
-    NSString* newDeviceToken = [[[[[deviceTokenData description]
-                                   stringByReplacingOccurrencesOfString:@"<"withString:@""]
-                                  stringByReplacingOccurrencesOfString:@">" withString:@""]
-                                 stringByReplacingOccurrencesOfString: @" " withString: @""] uppercaseString];
-    return newDeviceToken;
+    const char *data = [deviceTokenData bytes];
+    NSMutableString *newDeviceToken = [NSMutableString string];
+    
+    for (NSUInteger i = 0; i < [deviceTokenData length]; i++) {
+        [newDeviceToken appendFormat:@"%02.2hhx", data[i]];
```
Comment (brannon) [inline src/WindowsAzureMessaging/WindowsAzureMessaging/SBNotificationHub.m]: The previous code produced a token that was all uppercase (via the `uppercaseString` method).

The string format should be `%02.2hhX`.

**Comment by kross51** [comment]
Hi all, when I was previously googling around some posters said that the solution (similar to your guys' have) for converting NSData to string for deviceTokenData would work but something like the below would be more efficient or better; I don't remember.  Just throwing it out there in case what they said is valid and you wanted to implement the below instead:

NSUInteger bytesCount = deviceTokenData.length;
    if (bytesCount) {
        const char *hexChars = "0123456789ABCDEF";
        const unsigned char *dataBuffer = deviceTokenData.bytes;
        char *chars = malloc(sizeof(char) * (bytesCount * 2 + 1));
        if (chars == NULL) {
            // malloc returns null if attempting to allocate more memory than the system can provide. Thanks Cœur
            //[NSException raise:@"NSInternalInconsistencyException" format:@"Failed to allocate more memory" arguments:nil];
            return nil;
        }
        char *s = chars;
        for (unsigned i = 0; i < bytesCount; ++i) {
            *s++ = hexChars[((*dataBuffer & 0xF0) >> 4)];
            *s++ = hexChars[(*dataBuffer & 0x0F)];
            dataBuffer++;
        }
        *s = '\0';
        NSString *hexString = [NSString stringWithUTF8String:chars];
        free(chars);
        return hexString;
    }

**Comment by brannon** [comment]
@kross51 thanks! The implementation you shared is probably more efficient, but also less safe. Since the device tokens are only converted to strings periodically (when the token changes), I prefer to go with the safer, more straightforward implementation.

**Comment by brannon** [comment]
@iTOYS I'm concerned that this doesn't fix the issue as reported in #30. The call stack of the crash suggests that something is failing while generating the SharedAccessSignature.

I'm fine with merging this change as-is, but I want to make sure that we fix the crash. Were we able to reproduce the crash?

**Comment by arctouch-pedrocosta** [comment]
@brannon when a new release will be created?

**Pull request merged**